### PR TITLE
john-jumbo-devel: update to 20231220; john-jumbo*: works with py-scapy-devel

### PR DESCRIPTION
**Pull request by catap**
###### Type(s)

- [x] enhancement

###### Tested on

macOS 12.7.1 21G920 x86_64
Xcode 14.2 14C18

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? 
- [x] checked your Portfile with `port lint --nitpick`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?

### CODE REVIEW

File: sysutils/john/Portfile [hunk 1]
```diff
@@ -167,9 +167,12 @@ if {${subport} in [list "john-jumbo" "john-jumbo-devel"]} {
...
         depends_run-append      port:py${python3_version}-asn1 \
                                 port:py${python3_version}-cbor2 \
+                                port:py${python3_version}-hanko \
                                 port:py${python3_version}-lxml \
-                                port:py${python3_version}-pycryptodome \
-                                port:py${python3_version}-scapy
+                                port:py${python3_version}-pycryptodome
+
+        # allow installing both: py-scrapy and py-scrapy-devel
+        depends_run-append      path:${frameworks_dir}/Python.framework/Versions/python${python3_branch}/site-packages/scrapy/__init__.py:py${python3_version}-scapy
```
Comment (catap) [inline sysutils/john/Portfile:175]: @reneeotten can it be cleaner?

Comment (reneeotten) [reply sysutils/john/Portfile]: probably not unless you use the "python PG"

Comment (catap) [reply sysutils/john/Portfile]: Not sure that it help, because I use two different version of python at the same time here.

Comment (reneeotten) [reply sysutils/john/Portfile]: that should help. The only thing you need to check for is, for example,whether either py311-scrapy or py311-scrapy-devel is installed. (Of course you shouldn't hardcode the "3.11" in the check, that should rely on a version of ${python3_version} that contains a ".")

Comment (catap) [reply sysutils/john/Portfile]: I like that, thank!

Comment (catap) [reply sysutils/john/Portfile]: I'll care of remaining  scapy dependencies as part of https://github.com/macports/macports-ports/pull/21853

**Comment by catap** [comment]
Thanks, Renee!

**Pull request merged**